tqdm/_version.py

```diff
@@ -15,18 +11,35 @@
 
 
 # auto -extra based on commit hash (if not tagged as release)
-if (_sh is not None) and (len(version_info) < 4):  # pragma: no cover
-    def commit_hash(*args):
-        try:
-            res = _sh('git', 'log', '-n', '1', '--oneline', *args,
-                      stderr=STDOUT).lstrip().split()[0]
-            return None if res.startswith('fatal') else res
-        except:
-            return None
-
-    cur_hash = commit_hash()
-    if cur_hash is not None:
-        last_release = commit_hash('v' + __version__)
-
-        if (last_release is None) or (cur_hash not in last_release):
-            __version__ += '-' + cur_hash
+res = None
+scriptdir = os.path.dirname(__file__)
+gitdir = os.path.abspath(scriptdir + '/../.git')
+if os.path.isdir(gitdir):
+    # Open the HEAD file
+    with open(os.path.join(gitdir,'HEAD'), 'r') as fh_head:
+        res = fh_head.readline().strip()
+    # If we are in a branch, HEAD will point to a file containing the latest commit
+    if 'ref:' in res:
```

Comment:
In both of these branches, `res` is set. Yet above, `res` is set to another name. Two variables would be clearer.